atlas-api/src/main/java/org/atlasapi/query/v4/channel/ChannelQueryExecutor.java

```
@@ -133,46 +115,77 @@ public ChannelQueryExecutor(ChannelResolver resolver) {
-                                return resolved.getResources();
-                            }
-                        }
-                );
-        final ImmutableList<Channel> channels;
         try {
-            channels = ordering.immutableSortedCopy(Futures.get(
+            return Futures.get(
                     resolvedIterable,
                     1,
                     TimeUnit.MINUTES,
                     QueryExecutionException.class
-            ));
+            );
         } catch (Exception e) {
             throw new QueryExecutionException(e);
         }
+    }
+
+    private Ordering<? super Channel> ordering(String orderBy) {
+        Ordering<? super Channel> ordering = Ordering.allEqual();
+
+        if (Strings.isNullOrEmpty(orderBy)) {
+            return ordering;
+        }
 
-        ImmutableList<Channel> filteredChannels = ImmutableList.copyOf(Iterables.filter(
-                channels,
-                input -> {
-                    return query.getContext()
-                            .getApplicationSources()
-                            .isReadEnabled(input.getBroadcaster());
-                }
-        ));
-        return QueryResult.listResult(
-                query.getContext().getSelection().get().applyTo(
-                        filteredChannels
-                ),
-                query.getContext(),
-                filteredChannels.size()
-        );
+        switch (orderBy) {
+        case TITLE:
+            ordering = MoreOrderings.transformingOrdering(
+                    input -> Strings.nullToEmpty(input.getTitle())
+            );
+            break;
+        case TITLE_REVERSE:
+            ordering = MoreOrderings.transformingOrdering(
+                    input -> Strings.nullToEmpty(input.getTitle()),
+                    Ordering.<String>natural().reverse()
+            );
+            break;
+        default:
+            break;
```

This silently swallows invalid sort values. Generally, this should be an `IllegalArgumentException`, but depends on your case.